### PR TITLE
Issue #1746: Moving QUANTILE

### PR DESCRIPTION
**Pull request by hawkfish**
First pass at moving quantile implementation. There is more to do, but this has an important fix for the API by passing in `NULL` masks, which was a bug.

### CODE REVIEW

**Comment by hawkfish** [comment]
The latest change is just a perf change, but it impacts all windowed aggregates by removing unnecessary memory allocations from most segment tree computations. 

The perf improvement for windowed quantile (10M rows, fixed frame) is about 20%:

```shell
./build/reldebug/test/unittest test/sql/window/test_moving_quantile.test_slow  7.42s user 0.30s system 99% cpu 7.728 total
./build/reldebug/test/unittest test/sql/window/test_moving_quantile.test_slow  6.10s user 0.26s system 96% cpu 6.582 total
```

File: test/sql/window/test_quantile_window.test_slow [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# name: test/sql/window/test_quantile_window.test_slow
+# description: Test windowed QUANTILE at scale
+# group: [window]
+
+require vector_size 512
+
+#statement ok
+#PRAGMA enable_verification
```
Comment (Mytherin) [inline test/sql/window/test_quantile_window.test_slow:8]: Is this disabled by accident?

File: test/sql/window/test_moving_quantile.test_slow [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+create table rank100 as
+    select b % 100 as a, b
+    from (
+        select * from range(0, 10000000) tbl(b)
+    ) q;
+
+#   Fixed window
+statement ok
+prepare fwm as
+select sum(m)
```
Comment (Mytherin) [inline test/sql/window/test_moving_quantile.test_slow:18]: Can we check the result here?

File: test/sql/window/test_quantile_window.test_slow [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+#statement ok
+#create table rank1 as select 1 as a, b from range(0, 10000000) tbl(b) order by b desc;
+
+statement ok
+create table rank100 as select b % 100 as a, b from range(0, 10000000) tbl(b) order by b desc;
+
+statement ok
+select quantile(a, 0.5) over (
```
Comment (Mytherin) [inline test/sql/window/test_quantile_window.test_slow:17]: Can we check the result here?

File: test/sql/window/test_mode_window.test [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+3	1	1
+4	1	1
+5	1	1
+6	2	2
+7	2	2
+8	2	2
+9	3	2
+
+query III
+SELECT r, r/3, mode(r/3) over (order by r rows between 1 preceding and 3 following) FROM modes ORDER BY 1, 2
```
Comment (Mytherin) [inline test/sql/window/test_mode_window.test:48]: * Could we add a test with a bunch of scattered null values? e.g. you would order by column a, perform the quantile on column b, and column b is basically 50/50 null/not null.
* Perhaps a test with UNBOUNDED PRECEDING and a test with UNBOUNDED FOLLOWING? Perhaps also a test with both?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! The code looks good to me. Some more suggestions/ideas for the tests:

File: test/sql/window/test_quantile_window.test_slow (hunk 1 above)
Comment (hawkfish) [reply test/sql/window/test_quantile_window.test_slow:8]: Yes. More precisely, I was timing queries and wanted to remove overhead. I'll restore it.

Comment (hawkfish) [reply test/sql/window/test_quantile_window.test_slow:8]: After checking, I have removed it because it dramatically increases the run time.

File: test/sql/window/test_moving_quantile.test_slow (hunk 2 above)
Comment (hawkfish) [reply test/sql/window/test_moving_quantile.test_slow:18]: There are 10M values, but I will add a checksum.

File: test/sql/window/test_quantile_window.test_slow (hunk 1 above)
Comment (hawkfish) [reply test/sql/window/test_quantile_window.test_slow:8]: Upon review, this test file was an accident and should be removed. The other test file is more complete.

File: test/sql/window/test_mode_window.test (hunk 4 above)
Comment (hawkfish) [reply test/sql/window/test_mode_window.test:48]: This is `MODE` but yes we should hit them both.

Comment (hawkfish) [reply test/sql/window/test_mode_window.test:48]: These tests also drove a small bug in `Finalize` for `MODE` when the hash table is empty.

**Comment by Mytherin** [comment]
Thanks, looks great now!
